src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java

```diff
@@ -1515,15 +1519,32 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
         }
       };
 
-  private final Function<Booking, Mono<Booking>> checkBookingStatus =
+  private final Function<Booking, Mono<Booking>> checkCancelBookingStatus =
       booking -> {
         EnumSet<DocumentStatus> allowedDocumentStatuses =
-            EnumSet.of(DocumentStatus.RECE, DocumentStatus.PENU, DocumentStatus.CONF);
+            EnumSet.of(DocumentStatus.RECE, DocumentStatus.PENU, DocumentStatus.CONF, DocumentStatus.PENC);
         if (allowedDocumentStatuses.contains(booking.getDocumentStatus())) {
           return Mono.just(booking);
         } else
           return Mono.error(
               new UpdateException(
-                  "Cannot Cancel Booking that is not in status RECE, PENU or CONF"));
+                  "Cannot Cancel Booking that is not in status RECE, PENU, CONF or PENC"));
       };
+
+    private final Function<Booking, Mono<Booking>> checkUpdateBookingStatuses =
```

Comment:
I think we can consolidate this with the above method and keep the check on only `allowedDocumentStatuses` I don't think there is a specific need to check for `CANC`, @pimg what do you think?

Reply:
Do we not then somehow allow for cancelled bookings to be updated?

Reply:
I am quite sure Christian has said that once a Booking is Cancelled - then it cannot be modified... @pimg please also confirm...

Reply:
correct cancelled is an end state